include/rocksdb/db.h

```diff
@@ -1713,8 +1713,10 @@ class DB {
   }
 
   // Trace DB operations. Use EndTrace() to stop tracing.
-  virtual Status StartTrace(const TraceOptions& /*options*/,
-                            std::unique_ptr<TraceWriter>&& /*trace_writer*/) {
+  virtual Status StartTrace(
```

Comment:
According to Google Style guide which we follow, it's considered bad practice and not allowed to use default arguments on virtual functions: https://google.github.io/styleguide/cppguide.html#Default_Arguments.

I wonder whether we can avoid public API change. Just a thought, and haven't fully explored other options, how about adding a new overload at the layer of `DBImpl`, which is available in db_stress. Feel free to think about other better options to serve the needs here.

Reply:
Sounds good - I will avoid the public API change here.

Reply:
Since db_stress operate on DB instead of DBImpl, I decided to add another DB::start trace that takes the extra parameter and made the corresponding changes in DBImpl and StackDB.

Reply:
Fixed

Reply:
It's true that db_stress operates on `DB`, but `DBImpl` is also available. It's not like an external application in which you cannot do `static_cast_with_check<DBImpl>`. An example is https://github.com/facebook/rocksdb/blob/main/db_stress_tool/db_stress_test_base.cc#L2574.

Reply:
Fixed